tensorflow/compiler/xla/service/gpu/cusolver_context.cc

```diff
@@ -24,24 +24,51 @@ namespace {
 
 // Type traits to get CUDA complex types from std::complex<T>.
 template <typename T>
-struct CUDAComplexT {
+struct GpuComplexT {
   typedef T type;
 };
+#if !defined(TENSORFLOW_USE_ROCM)
+
+using gpuStream_t = cudaStream_t;
+
+#define gpusolverCreate cusolverDnCreate
+#define gpusolverSetStream cusolverDnSetStream
+#define gpusolverDestroy cusolverDnDestroy
+
 template <>
-struct CUDAComplexT<std::complex<float>> {
+struct GpuComplexT<std::complex<float>> {
   typedef cuComplex type;
 };
 template <>
-struct CUDAComplexT<std::complex<double>> {
+struct GpuComplexT<std::complex<double>> {
   typedef cuDoubleComplex type;
 };
 
+#else
+
+using gpuStream_t = hipStream_t;
+
+#define gpusolverCreate rocblas_create_handle
```

Comment:
Is this a function? Then `GpuSolverCreate` would be more consistent with other code?

Reply:
same as above